src/meta/src/manager/catalog/mod.rs

```diff
@@ -4078,6 +4101,13 @@ impl CatalogManager {
         let same_user = granted_by == revoke_by.id;
         if !revoke_by.is_super {
             for privilege in revoke_grant_privileges {
+                if Self::check_owner(
+                    catalog_core,
+                    privilege.object.as_ref().unwrap(),
+                    revoke_by.id,
+                )? {
+                    continue;
+                }
```

Comment:
We can move the check out of the iteration since the owner can always escape from the check.

```rust
if !revoke_by.is_super && !Self::check_owner(
 ...
) {
 ...
}
```

Reply:
Oh I make a mistake, `check_owner` needs to pin down the object stored in the privilege. Please ignore the comments 🤣.